iroh-net/src/endpoint.rs

```diff
@@ -631,37 +744,38 @@ impl Endpoint {
         }
     }
 
-    /// Inform the magic socket about addresses of the peer.
-    ///
-    /// This updates the magic socket's *netmap* with these addresses, which are used as candidates
-    /// when connecting to this peer (in addition to addresses obtained from a relay server).
+    /// Returns the DNS resolver used in this [`Endpoint`].
     ///
-    /// Note: updating the magic socket's *netmap* will also prune any connections that are *not*
-    /// present in the netmap.
+    /// See [`Builder::discovery`].
+    pub fn dns_resolver(&self) -> &DnsResolver {
+        self.msock.dns_resolver()
+    }
+
+    /// Returns the discovery mechanism, if configured.
     ///
-    /// # Errors
-    /// Will return an error if we attempt to add our own [`PublicKey`] to the node map.
-    pub fn add_node_addr(&self, node_addr: NodeAddr) -> Result<()> {
-        // Connecting to ourselves is not supported.
-        if node_addr.node_id == self.node_id() {
-            bail!(
-                "Adding our own address is not supported ({} is the node id of this node)",
-                node_addr.node_id.fmt_short()
-            );
-        }
-        self.msock.add_node_addr(node_addr);
-        Ok(())
+    /// See [`Builder::dns_resolver`].
+    pub fn discovery(&self) -> Option<&dyn Discovery> {
+        self.msock.discovery()
     }
 
-    /// Get a reference to the DNS resolver used in this [`Endpoint`].
-    pub fn dns_resolver(&self) -> &DnsResolver {
-        self.msock.dns_resolver()
+    /// Notifies the system of potential network changes.
+    ///
+    /// On many systems iroh is able to detect network changes by itself, however
+    /// some systems like android do not expose this functionality to native code.
+    /// Android does however provide this functionality to Java code.  This
+    /// function allows for notifying iroh of any potential network changes like
+    /// this.
+    ///
+    /// Even when the network did not change, or iroh was already able to detect
+    /// the network change itself, there is no harm in calling this function.
+    pub async fn network_change(&self) {
+        self.msock.network_change().await;
     }
 
-    /// Close the QUIC endpoint and the magic socket.
+    /// Closes the QUIC endpoint and the magic socket.
     ///
-    /// This will close all open QUIC connections with the provided error_code and reason. See
-    /// [quinn::Connection] for details on how these are interpreted.
+    /// This will close all open QUIC connections with the provided error_code and
+    /// reason. See [`quinn::Connection`] for details on how these are interpreted.
     ///
     /// It will then wait for all connections to actually be shutdown, and afterwards
     /// close the magic socket.
```

Comment:
:rainbow: `magic` :magic_wand:

Reply:
We still have a magic socket underneath.  It's the magic endpoint which no longer exists.  Do you think we should re-write this reference as well?  We left magic socket in a recent blog post as well.